components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/cache/ServiceProvideCacheInboundAuth.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.identity.application.mgt.internal.cache;
+
+import org.wso2.carbon.identity.application.common.cache.BaseCache;
+
+/**
+ * Cache to maintain the clint id - service provider name.
+ */
+public class ServiceProvideCacheInboundAuth extends
```

Comment:
Typo in the class name, missing letter 'r' in Provider word.

Reply:
Better to have Cache part at the end of the class name.
ie. ServiceProvideCacheInboundAuth ->
ServiceProviderByInboundAuthCache
ServiceProviderByIdCache
etc.